src/Umbraco.Infrastructure/Packaging/PackageDataInstallation.cs

```diff
@@ -1558,7 +1558,15 @@ public IReadOnlyList<ITemplate> ImportTemplates(IReadOnlyCollection<XElement> te
                 var masterElement = templateElement.Element("Master");
 
                 var existingTemplate = _fileService.GetTemplate(alias) as Template;
+
                 var template = existingTemplate ?? new Template(_shortStringHelper, templateName, alias);
+
+                // For new templates, use the serialized key if avaialble.
+                if (existingTemplate == null && templateElement.Element("Key") != null)
+                {
+                    template.Key = Guid.Parse(templateElement.Element("Key").Value);
+                }
```

Comment:
I've used the following in `UpdateContentTypesPropertyGroups()`:

https://github.com/umbraco/Umbraco-CMS/blob/2c530cf59bcb6b08a5a2b939831c9492a5f030ed/src/Umbraco.Infrastructure/Packaging/PackageDataInstallation.cs#L813-L816

I would suggest changing this to remove the duplicate `Element("Key")` calls and gracefully ignore invalid GUIDs:
```suggestion
                if (existingTemplate == null && Guid.TryParse(templateElement.Element("Key")?.Value, out var key)
                {
                    template.Key = key;
                }
```

Reply:
Sounds good, making this change.